posts/2022-09-01-java-19-22.0.0.10-beta.adoc

```diff
@@ -0,0 +1,330 @@
+
+The example below defines method `updateBell` within service implementation class `RestSci`.
+
+[source, java]
+----
+        package my.servlet.extensions;
+        ...    
+        public class RestSci implements javax.servlet.ServletContainerInitializer {
+           public RestSci() {}
+
+           public void updateBell(java.util.Map<String, String> properties) { 
+              // Collect properties 
+           }
+           ...
+        }
+----
+At runtime the server will make SPI packages visible only to a specialized classloader for library `ServletExtLib`. Applications can declare `commonLibRef="ServletExtLib"` in their `<classloader/>` configuration, but they will not see SPI packages. Regarding `<properties/>` the server will inject an unmodifiable map with `key=value` pairs `"servletName"="RestSciServlet"` and `"servletMapping"="/rest/*"` into all services when created during startup and after BELL configuration updates.
+
+
+The following steps describe how to bring up a BELL service that implements the REST Handler SPI interface `com.ibm.wsspi.rest.handler.RESTHandler`, which was impossible until now.
+
+=== Create the service resource and implementation class files
+
+The resource file is named for the SPI interface and declares the fully-qualified name of the implementation class. The minimal resource file content is shown below. Notice the file also declares an OSGi property required by the RESTHandler interface -- the entry starting with `#` -- that enables the REST Handler framework to register the service as a listener for a specified URL sub-root. In this example the URL sub-root is `/bellEP`.
+
+[source, java]
+----
+    #com.ibm.wsspi.rest.handler.root=/bellEP
+    your.org.rest.example.BellEndpoint
+----
+
+And here's a skeletal implementation of SPI interface `com.ibm.wsspi.rest.handler.RESTHandler`. The implementation defines method `updateBell` to receive BELL properties. The `updateBell` method is intended for non-destructive updates and must tolerate multiple calls on the same service instance.
+
+[source, java]
+----
+    package your.org.rest.example;
+    import com.ibm.wsspi.rest.handler.*;
+    ...
+    public class BellEndpoint implements com.ibm.wsspi.rest.handler.RESTHandler {
+        public BellEndpoint() {}
+        public void updateBell(java.util.Map<String, String> properties) {...}
+        @Override
+        public void handleRequest(RESTRequest request, RESTResponse response) {...}
+    }
+----
+
+Package the files into a JAR file, say `RestEpLib.jar`:
+
+[source, java]
+----
+    META-INF/services/com.ibm.wsspi.rest.handler.RESTHandler
+    your/org/rest/example/BellEndpoint.class
+----
+

```

Comment:
```suggestion
1. Create the service resource and implementation class files

   The resource file is named for the SPI interface and declares the fully-qualified name of the implementation class. The minimal resource file content is shown below. Notice the file also declares an OSGi property required by the RESTHandler interface -- the entry starting with `#` -- that enables the REST Handler framework to register the service as a listener for a specified URL sub-root. In this example the URL sub-root is `/bellEP`.

   [source, java]
   ----
   #com.ibm.wsspi.rest.handler.root=/bellEP
   your.org.rest.example.BellEndpoint
   ----

   And here's a skeletal implementation of SPI interface `com.ibm.wsspi.rest.handler.RESTHandler`. The implementation defines method `updateBell` to receive BELL properties. The `updateBell` method is intended for non-destructive updates and must tolerate multiple calls on the same service instance.

   [source, java]
   ----
   package your.org.rest.example;
   import com.ibm.wsspi.rest.handler.*;
   ...
   public class BellEndpoint implements com.ibm.wsspi.rest.handler.RESTHandler {
       public BellEndpoint() {}
       public void updateBell(java.util.Map<String, String> properties) {...}
       @Override
       public void handleRequest(RESTRequest request, RESTResponse response) {...}
   }
   ----

   Package the files into a JAR file, say `RestEpLib.jar`:

   [source, java]
   ----
   META-INF/services/com.ibm.wsspi.rest.handler.RESTHandler
   your/org/rest/example/BellEndpoint.class
   ----

```

Reply:
Basically, make this all part of Step 1 with indented paragraphs for everything in it.